### PR TITLE
New homepage header and hero

### PR DESCRIPTION
**Pull request by davidrapson**
This PR is _very_ work in-progress. Working through creating the new header and hero for the updated homepage (#75).

So far I have the _broad_ layout of the header and hero in place. There's plenty of refactoring and refinement to go to get this into a finished state, but wanted to get a PR up to show progress so far.

localhost-3000-

@mattandrews Don't worry too much about reviewing individual commits just yet as I'll aim to rebase / clean things up once I'm a bit further along. Shout if there's anything glaringly concerning so far though.

---

EDIT: Extracted the housekeeping changes into a separate PR here https://github.com/biglotteryfund/blf-alpha/pull/174

### CODE REVIEW

File: assets/sass/globals/accents.scss [hunk 1]
```diff
@@ -10,8 +10,6 @@
     border-top-style: solid;
 }
 
-.a--border-left {  }
```
Comment (mattandrews) [inline assets/sass/globals/accents.scss]: If a selector is deleted without ever having any content, was it ever really CSS?

File: views/pages/toplevel/home.njk [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 {% extends "../../layouts/main.njk" %}
 
+{% set bodyClass = "has-overlay" %}
```
Comment (mattandrews) [inline views/pages/toplevel/home.njk]: My only concern with this classname is that we also have the concept of an "overlay" (as opposed to `.overlay-text`), which is only used in a few places but is a modal-style box overlaid on top of the page when a successful "free material" order has been completed. This class sounds like it might be referring to that rather than the text for the hero? (see `overlay.scss` for more info!)

Comment (davidrapson) [reply views/pages/toplevel/home.njk]: Good shout, renamed this to the longer but hopefully more explicit `has-full-bleed-header`.

File: assets/sass/components/btn.scss [hunk 3]
```diff
@@ -172,4 +185,26 @@ $btnConf: (
...
-}
\ No newline at end of file
+}
+
+/* =========================================================================
+   Button Group
+   ========================================================================= */
+/* Group together a set of buttons with consistent spacing */
+
+.o-button-group {
```
Comment (davidrapson) [inline assets/sass/components/btn.scss]: Not sure we have something like this already, but this introduces the concept of a group of buttons to aid with spacing them consistently in a set. The `.o-` prefix is for "Objects" i.e. lower level structural components like buttons and icons. I tend to quite closely follow Harry Roberts naming conventions for this kind of thing, see: https://csswizardry.com/2015/03/more-transparent-ui-code-with-namespaces/ but can rename this prefix if there's existing terminology that better fits (e.g. is this more appropriately "scaffolding"?)

File: views/pages/toplevel/home.njk [hunk 4]
```diff
@@ -25,75 +28,73 @@
 {% block content %}
 
     {% set heroContent %}
-        {{ headers.overlayText('h2', "The Big Lottery Fund is the largest funder of community activities in the UK. We distribute money raised by players of the <a href=\"https://www.national-lottery.co.uk\">National Lottery</a>", 'overlay-text--has-links t2') }}
+        <div class="o-button-group">
+            <a class="btn btn--small [ accent--white a--btn ]"
```
Comment (davidrapson) [inline views/pages/toplevel/home.njk]: This square bracket thing is a method for grouping based on https://csswizardry.com/2014/05/grouping-related-classes-in-your-markup/ 

I was finding the distinction if different _types_ of classes hard to read so was aiming to differentiate between the unique module classes and modifiers. Happy to remove this if you find it to be too noisy.

File: views/pages/toplevel/home.njk [hunk 5]
```diff
@@ -25,75 +28,73 @@
...
+                <span class="accent--pink a--text">
+                    <span class="u-wide-only">Learn how to apply for funding under £10,000</span>
+                    <span class="u-not-wide">Apply for funding under £10,000</span>
+                </span>
+            </a>
+
+            <a class="btn btn--small [ accent--white a--btn ]"
+                href="{{ buildUrl('toplevel', 'over10k') }}">
+                <span class="accent--blue a--text">
+                    <span class="u-wide-only">Learn how to apply for funding over £10,000</span>
```
Comment (davidrapson) [inline views/pages/toplevel/home.njk]: Not sure how great the double text is here. Personally I'd prefer to use the shorter button label everywhere. I've made a note to bring that question up.

File: views/pages/toplevel/home.njk [hunk 6]
```diff
@@ -25,75 +28,73 @@
...
+                across the UK, raised by players of the <a href="https://www.national-lottery.co.uk/">National Lottery</a>.',
+            titleShort = 'The Big Lottery Fund distributes money raised by players of the
+                <a href="https://www.national-lottery.co.uk/">National Lottery</a> to communites across the UK.',
+            content = heroContent,
+            caption = 'Cloughmills Community Action, Grant £4,975*'
+        )
+    }}
 
     <article role="main">
-        <div class="inner">
```
Comment (davidrapson) [inline views/pages/toplevel/home.njk]: I'm scope creeping a little bit here my moving other parts of the homepage around to match the designs as part of this PR happy to pull these out into separate PRs.

File: assets/sass/components/btn.scss (hunk 3 above)
Comment (mattandrews) [reply assets/sass/components/btn.scss]: 👍  sensible choice. There's probably a juicy refactoring task here to rename some existing classes in this way (eg. see the hackery towards the end of `layout.scss`)

File: assets/sass/_utils.scss [hunk 7]
```diff
@@ -1,18 +1,41 @@
...
 }
 
+.u-hidden,
 .hidden {
     display: none;
 }
 
 // via http://simplyaccessible.com/article/three-pitfalls-text-alternatives/
+.u-visually-hidden,
 .visually-hidden {
```
Comment (mattandrews) [inline assets/sass/_utils.scss]: I think rather than supporting both naming conventions, let's just standardise on one of them (eg. the `.u-` prefix version) – same for the `.mobile-only` classes above this. I'm happy to do the search/replace job! Think it'd be nice to set a precedent here as we don't have any legacy templates to support or anything.

Comment (davidrapson) [reply assets/sass/_utils.scss]: Sure thing, that works for me if you're happy to standardise on this prefix for utility classes.

**Comment by davidrapson** [comment]
Going to give this a pass over in the morning to double check I've not knocked anything out on other pages then will merge this in for wider review from the rest of the team.
